dv_utils/datasets/contract.py

```diff
@@ -0,0 +1,110 @@
+            try:
+                logger.info(f"Create data contract object")
+                data_contract = DataContract(data_contract_str=data_contract_yaml)
+                self.data_contract=data_contract
+            except Exception as inst:
+                logger.error(Exception)
+                raise
+        else:  
+            logger.error(f"No data contract available")
+            raise RuntimeError("Unable to create data contract")
+
+    def __init_data_connector(self,data_descriptor: str):
+        try:
+            data_source_type=data_descriptor["settings"]["type"] 
+            logger.info(f"Initialise data connector for data source: type={data_source_type}")
+            #initialise datavillage connector to get access to data source access keys
+            #TODO add other type
+            if data_source_type=="s3":
+                config = s3.S3Configuration()
+                if self.data_connector_config_path!="":
+                    populate_configuration(self.data_descriptor_id,config,self.data_connector_config_path)
+                else:
+                    populate_configuration(self.data_descriptor_id,config)
+                self.connector = s3.S3Connector(config)
+            if data_source_type=="gcs":
+                config = gcs.GCSConfiguration()
+                if self.data_connector_config_path!="":
+                    populate_configuration(self.data_descriptor_id,config,self.data_connector_config_path)
+                else:
+                    populate_configuration(self.data_descriptor_id,config)
+                self.connector = gcs.GCSConnector(config)
+            if data_source_type=="azure":
+                config = azure.AZConfiguration()
+                if self.data_connector_config_path!="":
+                    populate_configuration(self.data_descriptor_id,config,self.data_connector_config_path)
+                else:
+                    populate_configuration(self.data_descriptor_id,config)
+                self.connector = azure.AZConnector(config)
+        except Exception as inst:
+            logger.error(Exception)
+            raise RuntimeError(f"Unable to initialise connector: {inst}")
+
+    def __data_descriptor_to_data_contract(self,data_descriptor: str):
+        try: 
+            if data_descriptor["kind"]=="Dataset":
+                #TODO
+                s=0
+            else:
+                data_contract_yaml=f'''
+dataContractSpecification: 0.9.3
+id: urn:datacontract:{data_descriptor["id"]}
+info:
+  title: {data_descriptor["name"]}
+  version: custom
+  description: {data_descriptor["description"]}
+'''
+            if data_descriptor["schema"]!="":
+                data_contract_yaml+=f'''
+models:
+  {data_descriptor["schema"]}
+'''
+            if data_descriptor["syntheticData"]!="":
+                data_contract_yaml+=f'''
+examples:
+  {data_descriptor["syntheticData"]}
+'''
+            return data_contract_yaml
```

Comment:
maybe we can use a library like [PyYaml](https://pyyaml.org/wiki/PyYAMLDocumentation) that can serialize a dict to yaml?

Reply:
Need to discuss :)